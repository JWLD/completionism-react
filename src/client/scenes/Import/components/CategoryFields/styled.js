import styled from 'styled-components'

import { mixins, vars } from 'style'

export const CheckboxWrap = styled.div`
  flex-grow: 1;
  flex-basis: 0;
  padding: 1.2rem 1.6rem;
  border-radius: 0.4rem;
  background-color: #444;

  :not(:last-child) {
    margin-right: 2rem;
  }
`

export const CheckboxLabel = styled.label`
  ${mixins.flexRight};

  flex-direction: row-reverse;
  width: 100%;
  color: white;
  font: 2.5rem DinMC;

  :not(:last-child) {
    margin-bottom: 0.4rem;
  }
`

export const CheckboxInput = styled.input`
  position: absolute;
  opacity: 0;
`

export const FakeInput = styled.div`
  ${mixins.flex};

  height: 2.2rem;
  width: 2.2rem;
  margin-right: 1rem;
  border-radius: 0.2rem;
  background-color: #333;
  color: #333;
  font-size: 1.8rem;

  ${CheckboxInput}:checked ~ & {
    background-color: ${vars.pos};
    color: white;
  }
`
