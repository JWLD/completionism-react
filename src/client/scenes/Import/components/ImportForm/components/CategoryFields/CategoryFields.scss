@import 'global';

$wrapWidth: 70rem;
$boxPadding: 2rem;

.import-page {
	.checkbox-wrap {
		flex-direction: column;
		flex-grow: 1;
		flex-basis: 0;
		justify-content: flex-start;
		padding: 1.2rem 1.6rem;
		border-radius: 0.4rem;
		background-color: $cmpGrey;

		&:not(:last-child) {
			margin-right: $boxPadding;
		}
	}

	.checkbox-wrap__label {
		flex-direction: row-reverse;
		justify-content: flex-end;
		width: 100%;
		color: white;
		font: 2.5rem DinMC;

		&:hover {
			cursor: pointer;
		}

		&:not(:last-child) {
			margin-bottom: 0.4rem;
		}
	}

	.checkbox-wrap__input {
		position: absolute;
		opacity: 0;

		&:checked ~ .checkbox-wrap__icon {
			background-color: $pos;
		}

		&:checked ~ .checkbox-wrap__icon svg {
			opacity: 1;
		}
	}

	.checkbox-wrap__icon {
		@include flex;
		height: 2.2rem;
		width: 2.2rem;
		margin-right: 1rem;
		border-radius: 0.2rem;
		background-color: #333;

		svg {
			font-size: 1.8rem;
			opacity: 0;
		}
	}
}
