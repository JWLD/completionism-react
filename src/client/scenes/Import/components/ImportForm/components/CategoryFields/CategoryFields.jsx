import React, { Component } from 'react';

import CATEGORIES from 'data/constants/categories';

import { CheckboxInput, CheckboxLabel, CheckboxWrap, FakeInput } from './styled';

class CategoryFields extends Component {
	renderInputRow({ key, name }) {
		const inputRow = (
			<CheckboxLabel key={key}>{name}
				<CheckboxInput type="checkbox" name={key} />
				<FakeInput className="fa fa-check" />
			</CheckboxLabel>
		);

		return inputRow;
	}

	renderInputRows(categories) {
		const inputRows = categories.map(category => (category.battleNet
			? this.renderInputRow(category)
			: null));

		return inputRows;
	}

	renderCheckboxColumn(type) {
		const typeCategories = CATEGORIES[type];

		const checkboxColumn = (
			<CheckboxWrap key={type}>
				{this.renderInputRows(typeCategories)}
			</CheckboxWrap>
		);

		return checkboxColumn;
	}

	renderCheckboxColumns() {
		const checkboxColumns = Object.keys(CATEGORIES).map(type => this.renderCheckboxColumn(type));

		return checkboxColumns;
	}

	render() {
		return this.renderCheckboxColumns();
	}
}

export default CategoryFields;
