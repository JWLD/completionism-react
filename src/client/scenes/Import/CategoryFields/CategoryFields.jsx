import React, { Component } from 'react';
import { Field } from 'redux-form';

import CATEGORIES from 'constants/categories';

import { GenericField } from 'components/ReduxFields';
import * as SC from './styled';

class CategoryFields extends Component {
	renderInputRows(categories) {
		const inputRows = categories.map(({ battleNet, key, name }) => {
			if (!battleNet) return null;

			return (
				<SC.CheckboxLabel key={key}>{name}
					<Field
						component={GenericField}
						name={key}
						StyledComponent={SC.CheckboxInput}
						type="checkbox"
					/>
					<SC.FakeInput className="fa fa-check" />
				</SC.CheckboxLabel>
			);
		});

		return inputRows;
	}

	renderCheckboxColumns() {
		const checkboxColumns = Object.keys(CATEGORIES).map(category => (
			<SC.CheckboxWrap key={category}>
				{this.renderInputRows(CATEGORIES[category])}
			</SC.CheckboxWrap>
		));

		return checkboxColumns;
	}

	render() {
		return this.renderCheckboxColumns();
	}
}

export default CategoryFields;
