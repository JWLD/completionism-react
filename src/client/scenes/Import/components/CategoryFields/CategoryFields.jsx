import React, { Component } from 'react';
import FaCheck from 'react-icons/lib/fa/check';
import CATEGORIES from 'data/constants/categories';

import './CategoryFields.scss';

class CategoryFields extends Component {
	changeCats(e) {
		const val = e.target.name;
		const arr = this.state.categories;

		if (arr.includes(val)) {
			arr.splice(arr.indexOf(val), 1);
		} else {
			arr.push(val);
		}

		this.setState({ categories: arr });
	}

	renderCats() {
		const cats = Object.keys(CATEGORIES).map(sub => (
			<div key={sub} className="cat-check-wrap">
				{CATEGORIES[sub].map((cat) => {
					if (!cat.battleNet) return null;

					return (
						<label key={cat.key}>{cat.name}
							<input type="checkbox" name={cat.key} onChange={this.changeCats} />
							<span>
								<FaCheck />
							</span>
						</label>
					);
				})}
			</div>
		));

		return cats;
	}

	render() {
		return (
			<section className="cat-sctn">
				<h2>Select Categories</h2>

				<div>
					{this.renderCats()}
				</div>
			</section>
		);
	}
}

export default CategoryFields;
