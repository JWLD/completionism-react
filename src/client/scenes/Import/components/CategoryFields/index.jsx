import React from 'react'
import { Field } from 'redux-form'

import { CATEGORIES_BY_SECTION as CATEGORIES } from 'constants/categories'

import * as SC from 'CategoryFields/styled'
import { CheckboxField } from 'formFields'

const CategoryFields = () => {
  const renderInputRows = categories => {
    return categories.map(({ battleNet, key, name }) => {
      if (!battleNet) return null

      return (
        <SC.CheckboxLabel key={key}>
          {name}
          <Field
            component={CheckboxField}
            name={key}
            StyledComponent={SC.CheckboxInput}
          />
          <SC.FakeInput className="fa fa-check" />
        </SC.CheckboxLabel>
      )
    })
  }

  const renderCheckboxColumns = () => {
    return Object.keys(CATEGORIES).map(category => (
      <SC.CheckboxWrap key={category}>
        {renderInputRows(CATEGORIES[category])}
      </SC.CheckboxWrap>
    ))
  }

  return renderCheckboxColumns()
}

export default CategoryFields
