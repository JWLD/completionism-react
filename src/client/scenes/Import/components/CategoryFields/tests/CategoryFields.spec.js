import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import CategoryFields from 'CategoryFields'

Enzyme.configure({ adapter: new Adapter() })

describe('<CategoryFields />', () => {
  it('renders correctly', () => {
    const component = shallow(<CategoryFields />)

    expect(component).toMatchSnapshot()
  })
})
