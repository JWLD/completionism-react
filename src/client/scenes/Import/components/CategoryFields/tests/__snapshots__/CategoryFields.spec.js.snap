// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryFields /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategoryFields />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <CheckboxWrap>
      <CheckboxLabel>
        Mounts
        <Field
          StyledComponent={[Function]}
          component={[Function]}
          name="mounts"
        />
        <FakeInput
          className="fa fa-check"
        />
      </CheckboxLabel>
      <CheckboxLabel>
        Pets
        <Field
          StyledComponent={[Function]}
          component={[Function]}
          name="pets"
        />
        <FakeInput
          className="fa fa-check"
        />
      </CheckboxLabel>
    </CheckboxWrap>,
    <CheckboxWrap>
      <CheckboxLabel>
        Alchemy
        <Field
          StyledComponent={[Function]}
          component={[Function]}
          name="alchemy"
        />
        <FakeInput
          className="fa fa-check"
        />
      </CheckboxLabel>
      <CheckboxLabel>
        Blacksmithing
        <Field
          StyledComponent={[Function]}
          component={[Function]}
          name="blacksmithing"
        />
        <FakeInput
          className="fa fa-check"
        />
      </CheckboxLabel>
      <CheckboxLabel>
        Enchanting
        <Field
          StyledComponent={[Function]}
          component={[Function]}
          name="enchanting"
        />
        <FakeInput
          className="fa fa-check"
        />
      </CheckboxLabel>
      <CheckboxLabel>
        Engineering
        <Field
          StyledComponent={[Function]}
          component={[Function]}
          name="engineering"
        />
        <FakeInput
          className="fa fa-check"
        />
      </CheckboxLabel>
      <CheckboxLabel>
        Inscription
        <Field
          StyledComponent={[Function]}
          component={[Function]}
          name="inscription"
        />
        <FakeInput
          className="fa fa-check"
        />
      </CheckboxLabel>
      <CheckboxLabel>
        Jewelcrafting
        <Field
          StyledComponent={[Function]}
          component={[Function]}
          name="jewelcrafting"
        />
        <FakeInput
          className="fa fa-check"
        />
      </CheckboxLabel>
      <CheckboxLabel>
        Leatherworking
        <Field
          StyledComponent={[Function]}
          component={[Function]}
          name="leatherworking"
        />
        <FakeInput
          className="fa fa-check"
        />
      </CheckboxLabel>
      <CheckboxLabel>
        Tailoring
        <Field
          StyledComponent={[Function]}
          component={[Function]}
          name="tailoring"
        />
        <FakeInput
          className="fa fa-check"
        />
      </CheckboxLabel>
    </CheckboxWrap>,
    <CheckboxWrap>
      <CheckboxLabel>
        Cooking
        <Field
          StyledComponent={[Function]}
          component={[Function]}
          name="cooking"
        />
        <FakeInput
          className="fa fa-check"
        />
      </CheckboxLabel>
      <CheckboxLabel>
        First Aid
        <Field
          StyledComponent={[Function]}
          component={[Function]}
          name="firstaid"
        />
        <FakeInput
          className="fa fa-check"
        />
      </CheckboxLabel>
      <CheckboxLabel>
        Mining
        <Field
          StyledComponent={[Function]}
          component={[Function]}
          name="mining"
        />
        <FakeInput
          className="fa fa-check"
        />
      </CheckboxLabel>
    </CheckboxWrap>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <CheckboxWrap>
        <CheckboxLabel>
          Mounts
          <Field
            StyledComponent={[Function]}
            component={[Function]}
            name="mounts"
          />
          <FakeInput
            className="fa fa-check"
          />
        </CheckboxLabel>
        <CheckboxLabel>
          Pets
          <Field
            StyledComponent={[Function]}
            component={[Function]}
            name="pets"
          />
          <FakeInput
            className="fa fa-check"
          />
        </CheckboxLabel>
      </CheckboxWrap>,
      <CheckboxWrap>
        <CheckboxLabel>
          Alchemy
          <Field
            StyledComponent={[Function]}
            component={[Function]}
            name="alchemy"
          />
          <FakeInput
            className="fa fa-check"
          />
        </CheckboxLabel>
        <CheckboxLabel>
          Blacksmithing
          <Field
            StyledComponent={[Function]}
            component={[Function]}
            name="blacksmithing"
          />
          <FakeInput
            className="fa fa-check"
          />
        </CheckboxLabel>
        <CheckboxLabel>
          Enchanting
          <Field
            StyledComponent={[Function]}
            component={[Function]}
            name="enchanting"
          />
          <FakeInput
            className="fa fa-check"
          />
        </CheckboxLabel>
        <CheckboxLabel>
          Engineering
          <Field
            StyledComponent={[Function]}
            component={[Function]}
            name="engineering"
          />
          <FakeInput
            className="fa fa-check"
          />
        </CheckboxLabel>
        <CheckboxLabel>
          Inscription
          <Field
            StyledComponent={[Function]}
            component={[Function]}
            name="inscription"
          />
          <FakeInput
            className="fa fa-check"
          />
        </CheckboxLabel>
        <CheckboxLabel>
          Jewelcrafting
          <Field
            StyledComponent={[Function]}
            component={[Function]}
            name="jewelcrafting"
          />
          <FakeInput
            className="fa fa-check"
          />
        </CheckboxLabel>
        <CheckboxLabel>
          Leatherworking
          <Field
            StyledComponent={[Function]}
            component={[Function]}
            name="leatherworking"
          />
          <FakeInput
            className="fa fa-check"
          />
        </CheckboxLabel>
        <CheckboxLabel>
          Tailoring
          <Field
            StyledComponent={[Function]}
            component={[Function]}
            name="tailoring"
          />
          <FakeInput
            className="fa fa-check"
          />
        </CheckboxLabel>
      </CheckboxWrap>,
      <CheckboxWrap>
        <CheckboxLabel>
          Cooking
          <Field
            StyledComponent={[Function]}
            component={[Function]}
            name="cooking"
          />
          <FakeInput
            className="fa fa-check"
          />
        </CheckboxLabel>
        <CheckboxLabel>
          First Aid
          <Field
            StyledComponent={[Function]}
            component={[Function]}
            name="firstaid"
          />
          <FakeInput
            className="fa fa-check"
          />
        </CheckboxLabel>
        <CheckboxLabel>
          Mining
          <Field
            StyledComponent={[Function]}
            component={[Function]}
            name="mining"
          />
          <FakeInput
            className="fa fa-check"
          />
        </CheckboxLabel>
      </CheckboxWrap>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
